sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -117,4 +118,47 @@ impl BlobClient {
             .await?;
         Ok(response)
     }
+
+    pub async fn commit_block_list(
+        &self,
+        blocks: RequestContent<BlockLookupList>,
+        options: Option<BlobBlockBlobClientCommitBlockListOptions<'_>>,
+    ) -> Result<Response<()>> {
+        let response = self
+            .client
+            .get_blob_block_blob_client(self.container_name.clone(), self.blob_name.clone())
+            .commit_block_list(blocks, options)
+            .await?;
+        Ok(response)
+    }
+
+    // Currently blocked by generated code, uncomment when we can consume newest definition
+    // pub async fn get_block_list(
+    //     &self,
+    //     list_type: BlockListType,
+    //     options: Option<BlobBlockBlobClientGetBlockListOptions<'_>>,
+    // ) -> Result<Response<BlockList>> {
+    //     let response = self
+    //         .client
+    //         .get_blob_block_blob_client(self.container_name.clone(), self.blob_name.clone())
+    //         .get_block_list(list_type, options)
+    //         .await?;
+    //     Ok(response)
+    // }
```

Comment:
This was fully working E2E against a forked copy of the generated code that contained the necessary fixes. This will be fixed when we can regenerate and get the necessary generated code changes (emitter currently is broken by newest `.tsp` definition).

Same comment applies to `get_service_properties` for `BlobServiceClient`.

Reply:
Removed from this PR to avoid confusion. Will introduce when the generated code backing is updated.